location/models.py

```diff
@@ -1,13 +1,34 @@
 from django.db import models
+from django.core.validators import RegexValidator
+
+valid_name = RegexValidator(r'^[&0-9a-zA-Z\s_\.\,-]+$',
+                            'Only alphanumeric characters, and _,.-_& are allowed.')
 
 
 class Location(models.Model):
-    name = models.CharField(max_length=40)
-    city = models.CharField(max_length=20)
-    street = models.CharField(max_length=20)
+    name = models.CharField(max_length=40, unique=True, validators=[valid_name])
+    city = models.CharField(max_length=20, validators=[valid_name])
+    street = models.CharField(max_length=20, validators=[valid_name])
     street_number = models.PositiveIntegerField()
     indoor = models.BooleanField()
     description = models.CharField(max_length=255)
 
     def __str__(self):
         return self.name
+
+    def update(self, name=None, city=None, street=None, street_number=None, indoor=None, description=None):
+        temp_location = Location.objects.get(id=self.id)
+        if name is not None:
+            temp_location.name = name
```

Comment:
I think it should be the same as using self
```suggestion
        if name is not None:
            self.name = name
            ...
```